### PR TITLE
Whitelist Version Check for Non-Stable Release Trains for connectedk8s

### PR DESCRIPTION
**Pull request by kitnhl**
Skip checking operation version support for non-stable release trains.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az connectedk8s upgrade -g <rg_name> -n <cluster_name>
(should work for any version for dev/private/other-non-stable release trains, but only for v>1.1.14 for stable release train, the same applies for az connectedk8s update/enable-features/disable-features)

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/connectedk8s/azext_connectedk8s/custom.py [hunk 1]
```diff
@@ -2538,6 +2538,6 @@ def crd_cleanup_force_delete(kubectl_client_location, kube_config, kube_context)
 
 def check_operation_support(operation_name, agent_version):
     error_summary = 'This CLI version does not support {} for Agents older than v1.14'.format(operation_name)
-    if (version.parse(agent_version) < version.parse("1.14.0")):
+    if '-' not in agent_version and (version.parse(agent_version) < version.parse("1.14.0")):
```
Comment (NarayanThiru) [inline src/connectedk8s/azext_connectedk8s/custom.py:2541]: Add a comment here explaining why we are doing this.  Something like "We want to do this check only for Stable versions, which will be of the pattern '1.x.x' versus other internal versions e.g. '1.x.x-dev'"

File: src/connectedk8s/azext_connectedk8s/_utils.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ def pull_helm_chart(registry_path, chart_export_path, kube_config, kube_context,
     chart_version = registry_path.split(':')[1]
 
     if new_path:
-        if (version.parse(chart_version) < version.parse("1.14.0")):
+        if '-' not in chart_version and (version.parse(chart_version) < version.parse("1.14.0")):
```
Comment (NarayanThiru) [inline src/connectedk8s/azext_connectedk8s/_utils.py:121]: Here too add a comment explaining why we are doing this. Something like "We want to do this check only for Stable versions, which will be of the pattern '1.x.x' versus other internal versions e.g. '1.x.x-dev'"

**Review by NarayanThiru — CHANGES_REQUESTED** [review]

**Review by NarayanThiru — APPROVED** [review]

File: src/connectedk8s/azext_connectedk8s/_utils.py [hunk 3]
```diff
@@ -118,7 +118,8 @@ def pull_helm_chart(registry_path, chart_export_path, kube_config, kube_context,
     chart_version = registry_path.split(':')[1]
 
     if new_path:
-        if (version.parse(chart_version) < version.parse("1.14.0")):
+        # Version check for stable release train (chart_version will be in X.Y.Z format as opposed to X.Y.Z-NONSTABLE)
+        if '-' not in chart_version and (version.parse(chart_version) < version.parse("1.14.0")):
```
Comment (zhoxing-ms) [inline src/connectedk8s/azext_connectedk8s/_utils.py:122]: Please note that the CLI team has defined a unified new extension version upgrade rule, please take a look https://github.com/Azure/azure-cli/blob/dev/doc/extensions/versioning_guidelines.md#cli-extension-version-transition-table

File: src/connectedk8s/setup.py [hunk 4]
```diff
@@ -17,7 +17,7 @@
 # TODO: Confirm this is the right version number you want and it matches your
 # HISTORY.rst entry.
 
-VERSION = '1.6.2'
+VERSION = '1.6.3'
```
Comment (zhoxing-ms) [inline src/connectedk8s/setup.py:20]: Please note that the CLI team has defined a unified new extension version upgrade rule, please take a look https://github.com/Azure/azure-cli/blob/dev/doc/extensions/versioning_guidelines.md#cli-extension-version-transition-table

Comment (kitnhl) [reply src/connectedk8s/setup.py:20]: Thank you for the heads up! Confirming that this is indeed a bug fix and 1.6.3 stands as the appropriate next version

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**